crypto/kdf/hkdf.c

```diff
@@ -128,6 +133,21 @@ static int pkey_hkdf_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2)
 static int pkey_hkdf_ctrl_str(EVP_PKEY_CTX *ctx, const char *type,
                               const char *value)
 {
+    if (strcmp(type, "mode") == 0) {
+        int mode;
+
+        if (strcmp(value, "EXTRACT_AND_EXPAND") == 0)
```

Comment:
another possibility is to remove the 'mode' variable and have three inline calls that set the mode with the inline constant.  then you can remove the else's.   it's okay if you reject that idea.